### PR TITLE
Do not allow running darwin/amd64 minikube binary on darwin/arm64 systems

### PR DESCRIPTION
**Pull request by ilya-zuyev**
Fixes https://github.com/kubernetes/minikube/issues/10928

Before:

```
ilyaz@mymac --- z/minikube ‹master*➔ ?› » out/minikube-darwin-amd64  start
😄  minikube v1.19.0-beta.0 on Darwin 11.2
✨  Using the docker driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
💾  Downloading Kubernetes v1.20.2 preload ...
    > preloaded-images-k8s-v10-v1...: 491.71 MiB / 491.71 MiB  100.00% 5.45 MiB
    > gcr.io/k8s-minikube/kicbase...: 353.96 MiB / 353.96 MiB  100.00% 3.58 MiB
🔄  Restarting existing docker container for "minikube" ...
🐳  Preparing Kubernetes v1.20.2 on Docker 20.10.3 ...
💢  initialization failed, will try again: wait: /bin/bash -c "sudo env PATH=/var/lib/minikube/binaries/v1.20.2:$PATH kubeadm init --config /var/tmp/minikube/kubeadm.yaml  --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--var-lib-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap,Mem,SystemVerification,FileContent--proc-sys-net-bridge-bridge-nf-call-iptables": Process exited with status 139 from signal SEGV
stdout:

stderr:
qemu: uncaught target signal 11 (Segmentation fault) - core dumped


💣  Error starting cluster: wait: /bin/bash -c "sudo env PATH=/var/lib/minikube/binaries/v1.20.2:$PATH kubeadm init --config /var/tmp/minikube/kubeadm.yaml  --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--var-lib-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap,Mem,SystemVerification,FileContent--proc-sys-net-bridge-bridge-nf-call-iptables": Process exited with status 139 from signal SEGV
stdout:

stderr:
SIGSEGV: segmentation violation
PC=0x0 m=0 sigcode=0

goroutine 1 [running]:
qemu: uncaught target signal 11 (Segmentation fault) - core dumped


😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose

❌  Exiting due to GUEST_START: wait: /bin/bash -c "sudo env PATH=/var/lib/minikube/binaries/v1.20.2:$PATH kubeadm init --config /var/tmp/minikube/kubeadm.yaml  --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--var-lib-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap,Mem,SystemVerification,FileContent--proc-sys-net-bridge-bridge-nf-call-iptables": Process exited with status 139 from signal SEGV
stdout:

stderr:
SIGSEGV: segmentation violation
PC=0x0 m=0 sigcode=0
```

After:
```
ilyaz@mymac --- z/minikube ‹ilyaz/do_not_run_amd64_on_m1* ?› » out/minikube-darwin-amd64 start

❌  Exiting due to MK_WRONG_BINARY_M1: You are trying to run amd64 binary on M1 system. Please use darwin/arm64 binary instead (Download at https://github.com/kubernetes/minikube/releases/download/v1.19.0-beta.0/minikube-darwin-amd64.
```

### CODE REVIEW

File: pkg/minikube/notify/notify.go [hunk 1]
```diff
@@ -152,3 +152,16 @@ func getTimeFromFileIfExists(path string) time.Time {
 	}
 	return timeInFile
 }
+
+// DownloadURL returns a URL to get minikube binary version ver for platform os/arch
+func DownloadURL(ver, os, arch string) string {
```
Comment (medyagh) [inline pkg/minikube/notify/notify.go:157]: how about unit test for this ?

File: cmd/minikube/cmd/root.go [hunk 2]
```diff
@@ -93,6 +95,11 @@ func Execute() {
 		}
 	}
 
+	if runtime.GOOS == "darwin" && detect.IsAmd64M1Emulation() {
```
Comment (medyagh) [inline cmd/minikube/cmd/root.go:98]: allow ppl to do this if they specify "--force" (if they do --force only warn them and don't exit)

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/root.go (hunk 2 above)
Comment (ilya-zuyev) [reply cmd/minikube/cmd/root.go:98]: It's 100% segfault right now, do we still want to allow it?

File: pkg/minikube/notify/notify.go (hunk 1 above)
Comment (ilya-zuyev) [reply pkg/minikube/notify/notify.go:157]: Yep, will add one

File: cmd/minikube/cmd/root.go (hunk 2 above)
Comment (medyagh) [reply cmd/minikube/cmd/root.go:98]: in that case no...

**Comment by medyagh** [comment]
also PTAL at the lint

File: pkg/minikube/notify/notify.go (hunk 1 above)
Comment (ilya-zuyev) [reply pkg/minikube/notify/notify.go:157]: done

**Comment by ilya-zuyev** [comment]
> also PTAL at the lint

fixed

**Pull request merged**